packages/app/src/hooks/useRowWhere.tsx

```diff
@@ -66,6 +67,52 @@ export default function useRowWhere({
                 value,
                 chType,
               ]);
+            case JSDataType.JSON:
+              // Handle case for whole json object, ex: json
+              return SqlString.format(`lower(hex(MD5(toString(?))))=?`, [
+                SqlString.raw(valueExpr),
+                MD5(value).toString(),
+              ]);
+            case JSDataType.Dynamic:
+              // Handle case for json element, ex: json.c
+
+              // Currently we can't distinguish null or 'null'
+              if (value === 'null') {
+                return SqlString.format(`isNull(??)`, SqlString.raw(column));
+              }
+              if (
+                value.length > 1000 ||
+                column.length > 1000 ||
+                columJsonSplit.length >= 8
+              ) {
+                throw new Error(
+                  'Search value/object key too large, or nested too many level',
+                );
+              }
+              if (columJsonSplit.length >= 2) {
+                // case for json.a, really bad implementation
+                // reason doing this because at CH
+                // {json: {a: [1,2,3]}}
+                // json -> '{"a": ["1","2","3"]}}', json.a -> [1,2,3]
+                // update this once there is new version
+                let jsonExtractTemplate = `simpleJSONExtractRaw(toString(?), ?)`;
+                for (let i = 2; i < columJsonSplit.length; i++) {
+                  jsonExtractTemplate =
+                    'simpleJSONExtractRaw(' + jsonExtractTemplate + ', ?)';
+                }
```

Comment:
what do you think if we throw here without implementing the hack temporarily? since I'd image this query could take a significant performance hit (string casting and parsing json for each depth)

Reply:
Yes, and that is the reason I add limitation on nested level. I also think it is fine to throw, our original plan is just support `number` and `string` type in current JSON version. We can always revisit this part if needed or get any JSON update.

Reply:
Btw, we don't have to throw, it just can not compare some types correctly, like `object` or `Array(Number)`.

Reply:
I think its fine for this version